game-app/game-core/src/main/java/games/strategy/triplea/image/UnitImageFactory.java

```diff
@@ -229,7 +232,15 @@ public boolean hasImage(final ImageKey imageKey) {
                           Util.ensureImageLoaded(scaledImage);
                           images.put(imageKey, scaledImage);
                           return scaledImage;
-                        }));
+                        }))
+        .orElseGet(
+            () -> {
+              final URL missingUnitImageLocation =
+                  resourceLoader.getResource(FILE_NAME_BASE + "/missing_unit_image.png");
+              Image image = Toolkit.getDefaultToolkit().getImage(missingUnitImageLocation);
+              Util.ensureImageLoaded(image);
+              return image;
```

Comment:
Do you know if [ImageIO#read(java.net.URL)](https://docs.oracle.com/en/java/javase/11/docs/api/java.desktop/javax/imageio/ImageIO.html#read(java.net.URL)) does the same but shorter?

Reply:
There is another library out there that does load an image and guarantees it to be actually loaded - though I do not recall if that was `ImageIO` or another library.

Reply:
It does seem ImageIO was the library we are looking for, and returns a `BufferedImage` as well which avoids future conversions.